<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from iiwa_striker.urdf.xacro        | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="iiwa_striker">
  <material name="Black">
    <color rgba="0.0 0.0 0.0 1.0"/>
  </material>
  <material name="Blue">
    <color rgba="0.0 0.0 0.8 1.0"/>
  </material>
  <material name="Green">
    <color rgba="0.0 0.8 0.0 1.0"/>
  </material>
  <material name="Grey">
    <color rgba="0.4 0.4 0.4 1.0"/>
  </material>
  <material name="Orange">
    <color rgba="1.0 0.4235294117647059 0.0392156862745098 1.0"/>
  </material>
  <material name="Brown">
    <color rgba="0.8705882352941177 0.8117647058823529 0.7647058823529411 1.0"/>
  </material>
  <material name="Red">
    <color rgba="0.8 0.0 0.0 1.0"/>
  </material>
  <material name="White">
    <color rgba="1.0 1.0 1.0 1.0"/>
  </material>
  <material name="LightGrey">
    <color rgba="0.3 0.3 0.3 1.0"/>
  </material>
  <link name="world"/>
  <!--joint between {parent} and link_0-->
  <joint name="world_F_base_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="world"/>
    <child link="F_link_0"/>
  </joint>
  <link name="F_link_0">
    <inertial>
      <!--  Fixed -->
      <origin rpy="0 0 0" xyz="-0.0 0 0.0"/>
      <mass value="5"/>
      <inertia ixx="0.05" ixy="0" ixz="0" iyy="0.06" iyz="0" izz="0.03"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://iiwas_description/meshes/visual/link_0.stl"/>
      </geometry>
      <material name="Grey"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://iiwas_description/meshes/collision/link_0.stl"/>
      </geometry>
      <material name="Grey"/>
    </collision>
    <self_collision_checking>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <capsule length="0.25" radius="0.15"/>
      </geometry>
    </self_collision_checking>
  </link>
  <!-- joint between link_0 and link_1 -->
  <joint name="F_joint_1" type="revolute">
    <parent link="F_link_0"/>
    <child link="F_link_1"/>
    <origin rpy="0 0 0" xyz="0 0 0.1575"/>
    <axis xyz="0 0 1"/>
    <limit effort="320" lower="-2.9670597283903604" upper="2.9670597283903604" velocity="1.4835298641951802"/>
    <safety_controller k_position="100" k_velocity="80" soft_lower_limit="-2.8187067419708423" soft_upper_limit="2.8187067419708423"/>
    <!-- Identified -->
    <dynamics damping="0.290458" friction="0.378214"/>
  </joint>
  <link name="F_link_1">
    <inertial>
      <!-- Identified Joint 1 -->
      <mass value="3.854820"/>
      <origin rpy="0.0 0.0 0.0" value="-0.098492 0.000000 0.069462"/>
      <inertia ixx="0.055704" ixy="0.000000" ixz="0.000356" iyy="0.056488" iyz="0.000000" izz="0.001462"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://iiwas_description/meshes/visual/link_1.stl"/>
      </geometry>
      <material name="Orange"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://iiwas_description/meshes/collision/link_1.stl"/>
      </geometry>
      <material name="Orange"/>
    </collision>
  </link>
  <!-- joint between link_1 and link_2 -->
  <joint name="F_joint_2" type="revolute">
    <parent link="F_link_1"/>
    <child link="F_link_2"/>
    <origin rpy="1.5707963267948966   0 3.141592653589793" xyz="0 0 0.2025"/>
    <axis xyz="0 0 1"/>
    <limit effort="320" lower="-2.0943951023931953" upper="2.0943951023931953" velocity="1.4835298641951802"/>
    <safety_controller k_position="100" k_velocity="80" soft_lower_limit="-1.9896753472735356" soft_upper_limit="1.9896753472735356"/>
    <!-- Identified -->
    <dynamics damping="0.465142" friction="0.372973"/>
  </joint>
  <link name="F_link_2">
    <inertial>
      <!-- Identified Joint 2 -->
      <mass value="5.773797"/>
      <origin rpy="0.0 0.0 0.0" value="-0.000687 -0.029334 0.119974"/>
      <inertia ixx="0.025239" ixy="0.003213" ixz="-0.000263" iyy="0.003716" iyz="0.001421" izz="0.023538"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://iiwas_description/meshes/visual/link_2.stl"/>
      </geometry>
      <material name="Orange"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://iiwas_description/meshes/collision/link_2.stl"/>
      </geometry>
      <material name="Orange"/>
    </collision>
  </link>
  <!-- joint between link_2 and link_3 -->
  <joint name="F_joint_3" type="revolute">
    <parent link="F_link_2"/>
    <child link="F_link_3"/>
    <origin rpy="1.5707963267948966 0 3.141592653589793" xyz="0 0.2045 0"/>
    <axis xyz="0 0 1"/>
    <limit effort="176" lower="-2.9670597283903604" upper="2.9670597283903604" velocity="1.7453292519943295"/>
    <safety_controller k_position="100" k_velocity="80" soft_lower_limit="-2.8187067419708423" soft_upper_limit="2.8187067419708423"/>
    <!-- Identified -->
    <dynamics damping="0.1" friction="0.213255"/>
  </joint>
  <link name="F_link_3">
    <inertial>
      <!-- Identified Joint 3 -->
      <mass value="5.953188"/>
      <origin rpy="0.0 0.0 0.0" value="0.000160 0.063306 0.043455"/>
      <inertia ixx="0.016829" ixy="0.000286" ixz="0.000417" iyy="0.011761" iyz="-0.005257" izz="0.006149"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://iiwas_description/meshes/visual/link_3.stl"/>
      </geometry>
      <material name="Orange"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://iiwas_description/meshes/collision/link_3.stl"/>
      </geometry>
      <material name="Orange"/>
    </collision>
  </link>
  <!-- joint between link_3 and link_4 -->
  <joint name="F_joint_4" type="revolute">
    <parent link="F_link_3"/>
    <child link="F_link_4"/>
    <origin rpy="1.5707963267948966 0 0" xyz="0 0 0.2155"/>
    <axis xyz="0 0 1"/>
    <limit effort="176" lower="-2.0943951023931953" upper="2.0943951023931953" velocity="1.3089969389957472"/>
    <safety_controller k_position="100" k_velocity="80" soft_lower_limit="-1.9896753472735356" soft_upper_limit="1.9896753472735356"/>
    <!-- Identified -->
    <dynamics damping="0.170443" friction="0.327965"/>
  </joint>
  <link name="F_link_4">
    <inertial>
      <!-- Identified Joint 4 -->
      <mass value="3.448720"/>
      <origin rpy="0.0 0.0 0.0" value="-0.000066 0.028782 0.116070"/>
      <inertia ixx="0.019103" ixy="0.000081" ixz="-0.000175" iyy="0.016046" iyz="0.001274" izz="0.003551"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://iiwas_description/meshes/visual/link_4.stl"/>
      </geometry>
      <material name="Orange"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://iiwas_description/meshes/collision/link_4.stl"/>
      </geometry>
      <material name="Orange"/>
    </collision>
  </link>
  <!-- joint between link_4 and link_5 -->
  <joint name="F_joint_5" type="revolute">
    <parent link="F_link_4"/>
    <child link="F_link_5"/>
    <origin rpy="-1.5707963267948966 3.141592653589793 0" xyz="0 0.1845 0"/>
    <axis xyz="0 0 1"/>
    <limit effort="110" lower="-2.9670597283903604" upper="2.9670597283903604" velocity="2.2689280275926285"/>
    <safety_controller k_position="100" k_velocity="80" soft_lower_limit="-2.8187067419708423" soft_upper_limit="2.8187067419708423"/>
    <!-- Identified -->
    <dynamics damping="0.1" friction="0.536120"/>
  </joint>
  <link name="F_link_5">
    <inertial>
      <!-- Identified Joint 5 -->
      <mass value="4.007168"/>
      <origin rpy="0.0 0.0 0.0" value="-0.000036 0.065820 0.034909"/>
      <inertia ixx="0.009814" ixy="-0.000066" ixz="-0.000189" iyy="0.008728" iyz="-0.001553" izz="0.001388"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://iiwas_description/meshes/visual/link_5.stl"/>
      </geometry>
      <material name="Orange"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://iiwas_description/meshes/collision/link_5.stl"/>
      </geometry>
      <material name="Orange"/>
    </collision>
  </link>
  <!-- joint between link_5 and link_6 -->
  <joint name="F_joint_6" type="revolute">
    <parent link="F_link_5"/>
    <child link="F_link_6"/>
    <origin rpy="1.5707963267948966 0 0" xyz="0 0 0.2155"/>
    <axis xyz="0 0 1"/>
    <limit effort="40" lower="-2.0943951023931953" upper="2.0943951023931953" velocity="2.356194490192345"/>
    <safety_controller k_position="100" k_velocity="80" soft_lower_limit="-1.9896753472735356" soft_upper_limit="1.9896753472735356"/>
    <!-- Identified -->
    <dynamics damping="0.1" friction="0.212718"/>
  </joint>
  <link name="F_link_6">
    <inertial>
      <!-- Identified Joint 6 -->
      <mass value="2.415840"/>
      <origin rpy="0.0 0.0 0.0" value="0.000723 0.019241 0.073756"/>
      <inertia ixx="0.002321" ixy="0.000637" ixz="0.000045" iyy="0.000980" iyz="-0.000262" izz="0.002200"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://iiwas_description/meshes/visual/link_6.stl"/>
      </geometry>
      <material name="Orange"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://iiwas_description/meshes/collision/link_6.stl"/>
      </geometry>
      <material name="Orange"/>
    </collision>
  </link>
  <!-- joint between link_6 and link_7 -->
  <joint name="F_joint_7" type="revolute">
    <parent link="F_link_6"/>
    <child link="F_link_7"/>
    <origin rpy="-1.5707963267948966 3.141592653589793 0" xyz="0 0.081 0"/>
    <axis xyz="0 0 1"/>
    <limit effort="40" lower="-3.0543261909900763" upper="3.0543261909900763" velocity="2.356194490192345"/>
    <safety_controller k_position="100" k_velocity="80" soft_lower_limit="-2.901609881440573" soft_upper_limit="2.901609881440573"/>
    <!-- Identified -->
    <dynamics damping="0.1" friction="0.337016"/>
  </joint>
  <link name="F_link_7">
    <inertial>
      <!-- Identified Joint 7 -->
      <mass value="1.107339"/>
      <origin rpy="0.0 0.0 0.0" value="0.001207 0.001384 -0.000630"/>
      <inertia ixx="0.000765" ixy="0.000276" ixz="-0.000333" iyy="0.001042" iyz="0.000211" izz="0.000818"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://iiwas_description/meshes/visual/link_7.stl"/>
      </geometry>
      <material name="Grey"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://iiwas_description/meshes/collision/link_7.stl"/>
      </geometry>
      <material name="Grey"/>
    </collision>
  </link>
  <joint name="F_joint_ee" type="fixed">
    <parent link="F_link_7"/>
    <child link="F_link_ee"/>
    <origin rpy="0 0 0" xyz="0 0 0.07"/>
  </joint>
  <link name="F_link_ee">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="1e-6"/>
      <inertia ixx="0.001" ixy="0" ixz="0" iyy="0.001" iyz="0" izz="0.001"/>
    </inertial>
  </link>
  <!-- Link0 -->
  <gazebo reference="F_link_0">
    <material>Gazebo/Grey</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>0</selfCollide>
  </gazebo>
  <!-- Link1 -->
  <gazebo reference="F_link_1">
    <material>Gazebo/Orange</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>0</selfCollide>
  </gazebo>
  <!-- Link2 -->
  <gazebo reference="F_link_2">
    <material>Gazebo/Orange</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>0</selfCollide>
  </gazebo>
  <!-- Link3 -->
  <gazebo reference="F_link_3">
    <material>Gazebo/Orange</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>0</selfCollide>
  </gazebo>
  <!-- Link4 -->
  <gazebo reference="F_link_4">
    <material>Gazebo/Orange</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>0</selfCollide>
  </gazebo>
  <!-- Link5 -->
  <gazebo reference="F_link_5">
    <material>Gazebo/Orange</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>0</selfCollide>
  </gazebo>
  <!-- Link6 -->
  <gazebo reference="F_link_6">
    <material>Gazebo/Orange</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>0</selfCollide>
  </gazebo>
  <!-- Link7 -->
  <gazebo reference="F_link_7">
    <material>Gazebo/Grey</material>
    <mu1>0.2</mu1>
    <mu2>0.2</mu2>
    <selfCollide>0</selfCollide>
  </gazebo>
  <!-- Link7 -->
  <!--    <gazebo reference="${robot_name}_joint_ee">-->
  <!--      <preserveFixedJoint>1</preserveFixedJoint>-->
  <!--    </gazebo>-->
  <gazebo reference="F_joint_1">
    <implicitSpringDamper>true</implicitSpringDamper>
  </gazebo>
  <gazebo reference="F_joint_2">
    <implicitSpringDamper>true</implicitSpringDamper>
  </gazebo>
  <gazebo reference="F_joint_3">
    <implicitSpringDamper>true</implicitSpringDamper>
  </gazebo>
  <gazebo reference="F_joint_4">
    <implicitSpringDamper>true</implicitSpringDamper>
  </gazebo>
  <gazebo reference="F_joint_5">
    <implicitSpringDamper>true</implicitSpringDamper>
  </gazebo>
  <gazebo reference="F_joint_6">
    <implicitSpringDamper>true</implicitSpringDamper>
  </gazebo>
  <gazebo reference="F_joint_7">
    <implicitSpringDamper>true</implicitSpringDamper>
  </gazebo>
  <transmission name="F_tran_1">
    <robotNamespace>/F</robotNamespace>
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="F_joint_1">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="F_motor_1">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="F_tran_2">
    <robotNamespace>/F</robotNamespace>
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="F_joint_2">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="F_motor_2">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="F_tran_3">
    <robotNamespace>/F</robotNamespace>
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="F_joint_3">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="F_motor_3">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="F_tran_4">
    <robotNamespace>/F</robotNamespace>
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="F_joint_4">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="F_motor_4">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="F_tran_5">
    <robotNamespace>/F</robotNamespace>
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="F_joint_5">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="F_motor_5">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="F_tran_6">
    <robotNamespace>/F</robotNamespace>
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="F_joint_6">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="F_motor_6">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="F_tran_7">
    <robotNamespace>/F</robotNamespace>
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="F_joint_7">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="F_motor_7">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <joint name="F_striker" type="fixed">
    <parent link="F_link_ee"/>
    <child link="F_striker_base"/>
    <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
  </joint>
  <link name="F_striker_base">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0.156"/>
      <mass value="0.350"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
    <visual>
      <origin rpy="0 0.0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://air_hockey_description/meshes/striker/visual/EE_arm.stl"/>
      </geometry>
      <material name="LightGrey"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://air_hockey_description/meshes/striker/collision/EE_arm_collision.stl"/>
      </geometry>
    </collision>
  </link>
  <joint name="F_striker_joint_1" type="revolute">
    <parent link="F_striker_base"/>
    <child link="F_striker_joint_link"/>
    <origin rpy="0 0 0" xyz="0.0 0.0 0.515"/>
    <axis xyz="0 1 0"/>
    <limit effort="1" lower="-1.5707963267948966" upper="1.5707963267948966" velocity="3.1415926"/>
    <safety_controller k_position="100" k_velocity="10" soft_lower_limit="-2.094395102" soft_upper_limit="2.094395102"/>
    <dynamics damping="0.05"/>
  </joint>
  <link name="F_striker_joint_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0.0 0"/>
      <mass value="0.05"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
  </link>
  <joint name="F_striker_joint_2" type="revolute">
    <parent link="F_striker_joint_link"/>
    <child link="F_striker_mallet"/>
    <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
    <axis xyz="1 0 0"/>
    <limit effort="1" lower="-1.5707963267948966" upper="1.5707963267948966" velocity="3.1415926"/>
    <safety_controller k_position="100" k_velocity="10" soft_lower_limit="-2.094395102" soft_upper_limit="2.094395102"/>
    <dynamics damping="0.05"/>
  </joint>
  <link name="F_striker_mallet">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0.0 0.0645"/>
      <mass value="0.183"/>
      <inertia ixx="0.01" ixy="0.0" ixz="0.0" iyy="0.01" iyz="0.0" izz="0.01"/>
    </inertial>
    <visual>
      <origin rpy="0.0 0.0 0" xyz="0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://air_hockey_description/meshes/striker/visual/EE_mallet_foam.stl"/>
      </geometry>
      <material name="LightGrey"/>
    </visual>
    <collision>
      <origin rpy="0.0 0.0 0" xyz="0 0.0 0.0455"/>
      <geometry>
        <cylinder length="0.04" radius="0.04815"/>
      </geometry>
    </collision>
  </link>
  <joint name="F_striker_mallet_fix" type="fixed">
    <parent link="F_striker_mallet"/>
    <child link="F_striker_mallet_tip"/>
    <origin rpy="0 0 0" xyz="0.0 0.0 0.0645"/>
    <axis xyz="1 0 0"/>
  </joint>
  <!--  The striker frame is here defined -->
  <link name="F_striker_mallet_tip">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0.0"/>
      <mass value="0.05"/>
      <inertia ixx="0.001" ixy="0.0" ixz="0.0" iyy="0.001" iyz="0.0" izz="0.001"/>
    </inertial>
  </link>
  <gazebo reference="F_striker_base">
    <collision>
      <surface>
        <bounce>
          <restitution_coefficient>1.0</restitution_coefficient>
          <threshold>0.0</threshold>
        </bounce>
        <contact>
          <collide_bitmask>0x02</collide_bitmask>
          <ode>
            <kp>1e9</kp>
            <max_vel>10</max_vel>
          </ode>
        </contact>
      </surface>
    </collision>
    <material>Gazebo/Grey</material>
    <mu1>1.0</mu1>
    <mu2>1.0</mu2>
    <selfCollide>0</selfCollide>
  </gazebo>
  <gazebo reference="F_striker_joint_link">
    <collision>
      <surface>
        <bounce>
          <restitution_coefficient>1.0</restitution_coefficient>
          <threshold>0.0</threshold>
        </bounce>
        <contact>
          <collide_bitmask>0x02</collide_bitmask>
          <ode>
            <kp>1e9</kp>
            <max_vel>10</max_vel>
          </ode>
        </contact>
      </surface>
    </collision>
    <material>Gazebo/Grey</material>
    <mu1>1.0</mu1>
    <mu2>1.0</mu2>
    <selfCollide>0</selfCollide>
  </gazebo>
  <gazebo reference="F_striker_mallet">
    <collision>
      <surface>
        <bounce>
          <restitution_coefficient>1.0</restitution_coefficient>
          <threshold>0.0</threshold>
        </bounce>
        <contact>
          <collide_bitmask>0x02</collide_bitmask>
          <ode>
            <kp>1e9</kp>
            <max_vel>10</max_vel>
          </ode>
        </contact>
      </surface>
    </collision>
    <material>Gazebo/Grey</material>
    <mu1>1.0</mu1>
    <mu2>1.0</mu2>
    <selfCollide>0</selfCollide>
  </gazebo>
  <gazebo reference="F_striker_mallet_fix">
    <preserveFixedJoint>1</preserveFixedJoint>
  </gazebo>
  <transmission name="F_striker_tran_1">
    <robotNamespace>/F</robotNamespace>
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="F_striker_joint_1">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="F_striker_motor_1">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="F_striker_tran_2">
    <robotNamespace>/F</robotNamespace>
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="F_striker_joint_2">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="F_striker_motor_2">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <gazebo>
    <plugin filename="libuniversal_joint_plugin.so" name="univsersal_joint_plugin">
            </plugin>
  </gazebo>
  <gazebo>
    <plugin filename="libgazebo_ros_control.so" name="gazebo_ros_control">
      <robotSimType>iiwas_gazebo/GravityCompensationHWSim</robotSimType>
    </plugin>
  </gazebo>
</robot>

